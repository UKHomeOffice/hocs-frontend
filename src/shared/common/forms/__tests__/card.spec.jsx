import React from 'react';
import Card from '../card.jsx';
import { MemoryRouter } from 'react-router-dom';

const DEFAULT_PROPS = {
    count: 5,
    label: 'Test Item',
    url: '/'
};

describe('Card component', () => {
    it('should render with default props', () => {
        expect(
            render(
                <MemoryRouter>
                    <Card {...DEFAULT_PROPS} />
                </MemoryRouter>
            )
        ).toMatchSnapshot();
    });

    it('should render with footer when children passed in props', () => {
        expect(
            render(
                <MemoryRouter>
                    <Card {...DEFAULT_PROPS}>
                        Footer
                    </Card>
                </MemoryRouter>
            )
        ).toMatchSnapshot();
    });
});

