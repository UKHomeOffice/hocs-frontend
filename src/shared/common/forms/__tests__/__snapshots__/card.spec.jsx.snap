// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="card govuk-body govuk-grid-column-one-quarter"
      >
        <a
          class="card__body"
          href="/"
        >
          <span
            class="govuk-!-font-size-48"
          >
            5
          </span>
          <span
            class="govuk-!-font-size-19 govuk-!-font-weight-bold"
          >
            Test Item
          </span>
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="card govuk-body govuk-grid-column-one-quarter"
    >
      <a
        class="card__body"
        href="/"
      >
        <span
          class="govuk-!-font-size-48"
        >
          5
        </span>
        <span
          class="govuk-!-font-size-19 govuk-!-font-weight-bold"
        >
          Test Item
        </span>
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card component should render with footer when children passed in props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="card govuk-body govuk-grid-column-one-quarter"
      >
        <a
          class="card__body"
          href="/"
        >
          <span
            class="govuk-!-font-size-48"
          >
            5
          </span>
          <span
            class="govuk-!-font-size-19 govuk-!-font-weight-bold"
          >
            Test Item
          </span>
        </a>
        <div
          class="card__footer"
        >
          Footer
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="card govuk-body govuk-grid-column-one-quarter"
    >
      <a
        class="card__body"
        href="/"
      >
        <span
          class="govuk-!-font-size-48"
        >
          5
        </span>
        <span
          class="govuk-!-font-size-19 govuk-!-font-weight-bold"
        >
          Test Item
        </span>
      </a>
      <div
        class="card__footer"
      >
        Footer
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
